src/data/ast.ts

```diff
@@ -1,31 +1,69 @@
 import { parseFirst, Statement, SelectFromStatement, astMapper, toSql, ExprRef } from 'pgsql-ast-parser';
 
-export function sqlToStatement(sql: string): Statement {
-  const replaceFuncs: Array<{
-    startIndex: number;
-    name: string;
-    replacementName: string;
-  }> = [];
-  //default is a key word in this grammar, but it can be used in CH
-  const re = /(\$__|\$|default|settings)/gi;
+interface ReplacePart {
+  startIndex: number;
+  name: string;
+  replacementName: string;
+};
+type ReplaceParts = ReplacePart[];
+
+/**
+ * Replaces macro functions and keywords such as $__timeFilter() and "default"
+ */
+function replaceMacroFunctions(sql: string): [ReplaceParts, string] {
+  const replaceFuncs: ReplaceParts = [];
+  // default is a keyword in this grammar, but it can be used in CH
+  const keywordRegex = /(\$__|\$|default|settings)/gi;
   let regExpArray: RegExpExecArray | null;
-  while ((regExpArray = re.exec(sql)) !== null) {
+  while ((regExpArray = keywordRegex.exec(sql)) !== null) {
     replaceFuncs.push({ startIndex: regExpArray.index, name: regExpArray[0], replacementName: '' });
   }
 
-  //need to process in reverse so starting positions aren't effected by replacing other things
+  // need to process in reverse so starting positions aren't affected by replacing other things
   for (let i = replaceFuncs.length - 1; i >= 0; i--) {
     const si = replaceFuncs[i].startIndex;
     const replacementName = 'f' + (Math.random() + 1).toString(36).substring(7);
     replaceFuncs[i].replacementName = replacementName;
     // settings do not parse and we do not need information from them so we will remove them
-    if (replaceFuncs[i].name.toLowerCase() === "settings") {
-      sql = sql.substring(0, si)
+    if (replaceFuncs[i].name.toLowerCase() === 'settings') {
+      sql = sql.substring(0, si);
       continue;
     }
     sql = sql.substring(0, si) + replacementName + sql.substring(si + replaceFuncs[i].name.length);
   }
 
+  return [replaceFuncs, sql];
+}
+
+/**
+ * Replaces Grafana variables such as ${var} ${var.key} ${var.key:singlequote}
+ */
```

Comment:
let's add a link to appropriate section https://grafana.com/docs/grafana/latest/dashboards/variables/